api.go

```diff
@@ -432,23 +435,28 @@ func (client *Client) PublishVolume(volumeId string, initiatorName string) (stri
 		}
 	}
 	for _, hostName := range hostNames {
-		if hostName != initiatorName {
-			return "", status.Errorf(codes.FailedPrecondition, "volume %s is already attached to another node", volumeId)
+		for _, initiator := range initiators {
+			if hostName == initiator {
+				klog.Infof("volume %s is already mapped to initiator %s", volumeId, initiators)
+			}
 		}
 	}
 
-	lun, err := client.chooseLUN(initiatorName)
+	lun, err := client.chooseLUN(initiators)
 	if err != nil {
 		return "", err
 	}
 
 	klog.Infof("using LUN %d", lun)
 
-	if err = client.mapVolumeProcess(volumeId, initiatorName, lun); err != nil {
-		return "", err
+	for _, initiator := range initiators {
+		//TODO optimize this for fewer API calls
+		if err = client.mapVolumeProcess(volumeId, initiator, lun); err != nil {
+			return "", err
+		}
 	}
 
-	klog.Infof("successfully mapped volume (%s) for initiator (%s) using LUN (%d)", volumeId, initiatorName, lun)
+	klog.Infof("successfully mapped volume (%s) for initiator (%s) using LUN (%d)", volumeId, initiators, lun)
```

Comment:
This would have to be modified to handle mix of successful and failed initiator mappings.